platform/features/table/src/controllers/MCTTableController.js

```diff
@@ -51,22 +61,84 @@ define(
                 self.updateRows($scope.rows);
             };
 
+            /*
+             * Define watches to listen for changes to headers and rows.
+             */
             $scope.$watchCollection('filters', function () {
-                self.updateRows(self.$scope.rows);
+                self.updateRows($scope.rows);
             });
-            $scope.$watchCollection('headers', this.updateHeaders.bind(this));
-            $scope.$watchCollection('rows', this.updateRows.bind(this));
+            $scope.$watch('headers', this.updateHeaders.bind(this));
+            $scope.$watch('rows', this.updateRows.bind(this));
+
+            /*
+             * Listen for rows added individually (eg. for real-time tables)
+             */
+            $scope.$on('addRow', this.newRow.bind(this));
         }
 
         /**
-         * On scroll, calculate which rows indexes are visible and
-         * ensure that an equal number of rows are preloaded for
-         * scrolling in either direction.
+         * If auto-scroll is enabled, this function will scroll to the
+         * bottom of the page
+         * @private
+         */
+        MCTTableController.prototype.scrollToBottom = function() {
+            var self = this;
+
+            //Use timeout to defer execution until next digest when any
+            // pending UI changes have completed, eg. a new row in the table.
+            if (this.$scope.autoScroll) {
+                this.$timeout(function(){
+                        self.scrollable[0].scrollTop = self.scrollable[0].scrollHeight;
+                });
+            }
+        };
+
+        /**
+         * Handles a row add event. Rows can be added as needed using the
+         * `addRow` broadcast event.
+         * @private
+         */
+        MCTTableController.prototype.newRow = function (event, row) {
+            //Add row to the filtered, sorted list of all rows
+            if (this.filterRows([row]).length > 0) {
+                this.insertSorted(this.$scope.displayRows, row);
+            }
+
+            //Keep 'rows' synchronized as it provides the unsorted,
+            // unfiltered model for this view
+            if (!this.$scope.rows) {
+                this.$scope.rows = [];
+            }
+            this.$scope.rows.push(row);
```

Comment:
Thanks for the alternative approaches, interesting to see how other APIs approach this. 
`$scope.broadcast('add:row', index)` - I could live with this. An alternative approach is to do away with $scope.rows altogether, and just use
* `$broadcast('setRows', rows);` 
* `$broadcast('addRow', row);`

I'm inclined to stick with $scope.rows though in order to keep the contract with the directive clear (you set headers and rows in the same way - via attributes on the directive, and it's consistent with how you pass values in other directives).